ext/ddtrace_profiling_native_extension/private_vm_api_access.c

```diff
@@ -712,8 +726,29 @@ check_method_entry(VALUE obj, int can_be_svar)
 #ifndef NO_RACTORS
   // This API and definition are exported as a public symbol by the VM BUT the function header is not defined in any public header, so we
   // repeat it here to be able to use in our code.
-  bool rb_ractor_main_p_(void);
-  extern struct rb_ractor_struct *ruby_single_main_ractor;
+  #ifndef USE_RACTOR_INTERNAL_APIS_DIRECTLY
+    // Disable fast path for detecting multiple Ractors. Unfortunately this symbol is no longer visible on modern Ruby
+    // versions, so we need to do a bit more work.
+    struct rb_ractor_struct *ruby_single_main_ractor = NULL;
+
+    // Taken from upstream ractor.c at commit a1b01e7701f9fc370f8dff777aad6d39a2c5a3e3 (May 2023, Ruby 3.3.0-preview1)
```

Comment:
I noticed we only place the commit sha, would it be useful to have the full URL?

Reply:
Hmm... I haven't considered, because usually I leave this to be able to have a reference when comparing to earlier/later changes. So usually I don't find a lot of value in looking at this commit, but looking at others that happened before or after it.

On the other hand, it wouldn't hurt, so I don't mind adding it if you find it relevant.

Reply:
I am happy either way. I noticed because I have to go and look at the code for the ruby source and then search for the commit.

Of course, that was my flow. Others might have the code locally and simply checkout to that specific commit and check. 

So I leave it to you 😄

Reply:
Ok so for now I'll be lazy, but I promise to add it if we find a third person that looks at it and also finds it missing :)